src/java.base/share/classes/java/io/BufferedInputStream.java

```diff
@@ -242,21 +238,13 @@ public BufferedInputStream(InputStream in, int size) {
             throw new IllegalArgumentException("Buffer size <= 0");
         }
         initialSize = size;
-        if (getClass() == BufferedInputStream.class) {
-            // use internal lock and lazily create buffer when not subclassed
-            lock = InternalLock.newLockOrNull();
-            buf = EMPTY;
-        } else {
-            // use monitors and eagerly create buffer when subclassed
-            lock = null;
-            buf = new byte[size];
-        }
+        buf = new byte[size];
```

Comment:
This should probably still retain the lazy buffer array creation:
```suggestion
size;
        if (getClass() == BufferedInputStream.class) {
            // lazily create buffer when not subclassed
            buf = EMPTY;
        } else {
            // eagerly create buffer when subclassed
            buf = new byte[size];
        }
```

Reply:
Indeed.  This and that for BufferedOutputStream are for backward compatibility with subclasses.  We cannot control when subclasses access the `buf` field after the constructor call, so laziness was impossible.